src/services/services.ts

```diff
@@ -2072,7 +2079,8 @@ namespace ts {
             compilationSettings: CompilerOptions,
             scriptSnapshot: IScriptSnapshot,
             version: string,
-            acquiring: boolean): SourceFile {
+            acquiring: boolean,
+            scriptKind?: ScriptKind): SourceFile {
 
             const bucket = getBucketForCompilationSettings(compilationSettings, /*createIfMissing*/ true);

```

Comment:
just for the sake of brainstorming: is it possible / legal scenario for host to start treating some existing file differently, i.e. in one project some file is treated as `.ts` and in another - as `.js`. Previously it was not possible since we derive script kind from extension which is the part of file's unique id which stays the same what project includes the file - its path. Currently from the API perspective this situation is possible...